### PR TITLE
Load Samples app Pages within Frame

### LINKED ISSUE #4684: Load samples which are a Page in Frame
## What would you like to be added:

WinUI samples often utilize `OnNavigatedTo/From` and `Frame` properties of test pages. The Samples app currently just materializes the page without a `Frame`.

## Why is this needed:

Make porting test pages from WinUI easier.

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #4684

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

SamplesApp `Page` samples are instantiated using `Activator`.

## What is the new behavior?

`Page` samples are navigated to using a `Frame`. This ensures their `Page.Frame` property is not `null` and `OnNavigatedTo` method is called.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]

**Review by carldebilly — APPROVED** [review]

**Review by agneszitte-nventive — APPROVED** [review]

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Media/GeometryTests/UnoSamples_Tests.ClosedFigurePage.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public void ClosedPath()
 
 			using var screenshot = TakeScreenshot("Closed state");
 
-			ImageAssert.HasColorAt(screenshot, 50, 300, Color.Black);
+			ImageAssert.HasColorAt(screenshot, 25, 300, Color.Black);
```
Comment (nickrandolph) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Media/GeometryTests/UnoSamples_Tests.ClosedFigurePage.cs:28]: Is it worth having an explanation as to why this position was chosen to test the color (and why 50,300 wasn't good?)

**Review by nickrandolph — APPROVED** [review]

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Media/GeometryTests/UnoSamples_Tests.ClosedFigurePage.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Media/GeometryTests/UnoSamples_Tests.ClosedFigurePage.cs]: Reverted

**Pull request merged**